src/cloudai/workloads/ai_dynamo/ai_dynamo.sh

```
@@ -187,21 +154,104 @@ _parse_cli_pairs() {
+  local num_nodes="$1"
+  local exclude_nodelist="$2" 
+
+  # Handle zero nodes case
+  if [[ -z "$num_nodes" || "$num_nodes" -eq 0 ]]; then
+    echo ""
+    return
+  fi
+
+  local count=0
+  local nodelist=""
+  for node in $(echo $DYNAMO_NODELIST | cut -d',' -f1-${num_nodes}); do
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)